lib/sequenceserver/blast/job.rb

```diff
@@ -88,6 +89,10 @@ def efile
         File.join(dir, 'efile')
       end
 
+      def delete_from_dotdir
```

Comment:
I think this should be a class method on `Job` (superclass), if at all required. It's similar to `create` and `fetch` methods defined on `Job`.

In (modern) OOP, it's often a good idea to define instances only as data objects and define methods that work on the data objects, "the services", in another module. In Ruby, class methods often work well to define a "service object", especially if there's only one kind of service around the data object - in this case create, read, update, delete (CRUD).